### PR TITLE
Message Filters and Enums PR

### PR DESCRIPTION
**Pull request by cynthia-m**

### CODE REVIEW

File: code/src/main/java/com/onlinecontacttracing/messaging/LocalityResource.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package com.onlinecontacttracing.messaging;
+
+public enum LocalityResource implements HasEnglishTranslation, HasSpanishTranslation{
+  US("https://www.cdc.gov/coronavirus/2019-ncov/index.html", "Here's the national help link: ", "Aquí está el enlace de ayuda nacional: "),
```
Comment (bengaskagoogle) [inline code/src/main/java/com/onlinecontacttracing/messaging/LocalityResource.java]: We talked about this before I believe, but having the same phrase repeated 50 times here is not useful. Break it off into a single static variable for both English and Spanish. The Enum can just have the link which is unique for a given state.

File: code/src/main/java/com/onlinecontacttracing/messaging/filters/FileReader.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.FileInputStream;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+
+public class FileReader {
```
Comment (bengaskagoogle) [inline code/src/main/java/com/onlinecontacttracing/messaging/filters/FileReader.java:17]: javadoc

File: code/src/main/java/com/onlinecontacttracing/messaging/filters/ProfanityFlaggingFilter.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            return false;
+          }
+        }
+      }
+      return true;
+    
+    
+  }
+
+  public boolean checkIfWordIsSelfContained (String message, String word, int messageLength) {
```
Comment (bengaskagoogle) [inline code/src/main/java/com/onlinecontacttracing/messaging/filters/ProfanityFlaggingFilter.java]: Probably these should all be private. Everything that isn't specified in the interface most likely.

File: code/src/main/java/com/onlinecontacttracing/messaging/filters/ProfanityFlaggingFilter.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    message = message.replaceAll("1","i");
+    message = message.replaceAll("3","e");
+    message = message.replaceAll("4","a");
+    message = message.replaceAll("5","s");
+    message = message.replaceAll("6","g");
+    message = message.replaceAll("7","t");
+    message = message.replaceAll("8","b");
+    message = message.replaceAll("9","g");
+    message = message.replaceAll("@","a");
+    message = message.replaceAll("!","i");
```
Comment (bengaskagoogle) [inline code/src/main/java/com/onlinecontacttracing/messaging/filters/ProfanityFlaggingFilter.java:106]: Why are you calling it twice?

File: code/src/main/java/com/onlinecontacttracing/messaging/filters/ProfanityFlaggingFilter.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+          isSelfContainedWord = checkIfWordIsSelfContained(message, profanityIndicatorWord, messageLength);
+          if (isSelfContainedWord) {
+            //profane word exists and is its own word => does not pass filter
+            return false;
+          }
+        }
+      }
+      return true;
+    
+
```
Comment (bengaskagoogle) [inline code/src/main/java/com/onlinecontacttracing/messaging/filters/ProfanityFlaggingFilter.java]: White space. Give the whole thing a look over for it, please.

File: code/src/main/webapp/JavaScript/script.js [hunk 6]
```diff
@@ -174,9 +174,15 @@ function attachSignin(element, negativeUser) {
     localStorage.setItem('idToken', idToken.toString());
     const params = new URLSearchParams()
     params.append('idToken', idToken);
-    fetch(new Request('/authentication-test', {method: 'POST', body: params})).then(response => response.text()).then(url => window.location = url);
-
+    params.append('systemMessage', 'VERSION_1');
+    params.append('localityResource', 'US');
+    params.append('messageLanguage', 'SP');
+    console.log("here");
```
Comment (bengaskagoogle) [inline code/src/main/webapp/JavaScript/script.js]: Get rid of this debugging logic before merging.

File: code/src/main/webapp/JavaScript/script.js [hunk 7]
```diff
@@ -174,9 +174,15 @@ function attachSignin(element, negativeUser) {
...
+    console.log(params);
```
Comment (bengaskagoogle) [inline code/src/main/webapp/JavaScript/script.js]: Same here.

File: code/src/main/webapp/JavaScript/script.js [hunk 8]
```diff
@@ -174,9 +174,15 @@ function attachSignin(element, negativeUser) {
...
+    fetch(new Request('/send-messages', {method: 'POST', body: params})).then(console.log("cool"));
+    // url => window.location = url
+    console.log("cool2");
```
Comment (bengaskagoogle) [inline code/src/main/webapp/JavaScript/script.js]: And here

File: code/src/main/webapp/html/messageSendTest.html [hunk 9]
```diff
@@ -0,0 +1,14 @@
+
+<!DOCTYPE HTML>
```
Comment (bengaskagoogle) [inline code/src/main/webapp/html/messageSendTest.html]: I don't know if you still need this test. Feel free to leave it if you do.

File: code/src/main/webapp/JavaScript/script.js [hunk 10]
```diff
@@ -166,17 +166,19 @@ var startApp = negativeUser => {
...
+
     const params = new URLSearchParams()
     params.append('idToken', idToken);
-    fetch(new Request('/authentication-test', {method: 'POST', body: params})).then(response => response.text()).then(url => window.location = url);
-
+    params.append('systemMessage', 'VERSION_1');
+    params.append('localityResource', 'US');
+    params.append('messageLanguage', 'SP');
+    
+    fetch(new Request('/send-messages', {method: 'POST', body: params})).then(console.log("request satisfied"));
```
Comment (bengaskagoogle) [inline code/src/main/webapp/JavaScript/script.js]: Do we need this log?

File: code/src/main/webapp/WEB-INF/credentials.json [hunk 11]
```diff
@@ -0,0 +1 @@
+{"web":{"client_id":"1080865471187-u1vse3ccv9te949244t9rngma01r226m.apps.googleusercontent.com","project_id":"covid-catchers-fixed-gcp","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"LpbG3OQ9SqSckXfkWikxXFUn","redirect_uris":["https://covid-catchers-fixed-gcp.ue.r.appspot.com"],"javascript_origins":["https://8080-ac896ae1-5f0c-45b5-8dfe-19fa4d3d8699.us-east1.cloudshell.dev","https://covid-catchers-fixed-gcp.ue.r.appspot.com"]}}
```
Comment (bengaskagoogle) [inline code/src/main/webapp/WEB-INF/credentials.json]: Do we want to not check this in? We should confirm with grady but you usually don't want to check this into ever, ever.

File: code/src/main/webapp/WEB-INF/appengine-web.xml [hunk 12]
```diff
@@ -10,6 +10,8 @@
 
   <resource-files>
     <include path="/**.json" />
+    <include path="/**.p12" />
```
Comment (bengaskagoogle) [inline code/src/main/webapp/WEB-INF/appengine-web.xml:12]: I'm also not certain we want to check in the p12 file above. You usually don't check in binary files. What is it?

File: code/src/main/webapp/WEB-INF/credentials.json (hunk 11 above)
Comment (cynthia-m) [reply code/src/main/webapp/WEB-INF/credentials.json]: yup I will remove it

File: code/src/main/webapp/WEB-INF/appengine-web.xml (hunk 12 above)
Comment (cynthia-m) [reply code/src/main/webapp/WEB-INF/appengine-web.xml:12]: yup we don't want to, it's the credentials access file for the service account

**Review by bengaskagoogle — APPROVED** [review]

**Pull request merged**